src/music.c

```diff
@@ -1,44 +1,1127 @@
+    }
+    else
+    {
+        if((speed >>= 4) == 0)
+        {
+            speed = 1;
+        }
+    }
+
+    if (songIndex == STOP_SOUND_EFFECT) {
+        cVar1 = DisableInterrupts();
+        for(playerIndex2 = INDEX_SE1, musicPlayer1 = &gUnknown_3000FE8[0]; playerIndex2 < INDEX_SE6; playerIndex2++, musicPlayer1++)
+        {
+            if (musicPlayer1->songIndex != STOP_SOUND_EFFECT) {
+                if (IsMusicPlayerPlaying(playerIndex2)) {
+                    m4aMPlayFadeOut(gMPlayTable[playerIndex2].info,speed);
+                }
+                else {
+                    m4aMPlayStop(gMPlayTable[playerIndex2].info);
+                    musicPlayer1->unk0 = 0;
+                    musicPlayer1->songIndex = STOP_SOUND_EFFECT;
+                    musicPlayer1->volume = 0;
+                    musicPlayer1->unk6 = FALSE;
+                }
+            }
+        }
+        if (cVar1 != '\0') {
+            EnableInterrupts();
+        }
+    }
+    else if (IsSoundEffect(songIndex)) {
+        playerIndex = GetMusicPlayerIndex(songIndex);
+        musicPlayer = &gUnknown_3000FD8[playerIndex];
+        playerInfo = gMPlayTable[playerIndex].info;
+        cVar2 = DisableInterrupts();
+        if (musicPlayer->songIndex != STOP_SOUND_EFFECT) {
+            if (IsMusicPlayerPlaying(playerIndex)) {
+                m4aMPlayFadeOut(playerInfo,speed);
+            }
+            else {
+                m4aMPlayStop(playerInfo);
+                musicPlayer->unk0 = 0;
+                musicPlayer->songIndex = STOP_SOUND_EFFECT;
+                musicPlayer->volume = 0;
+                musicPlayer->unk6 = FALSE;
+            }
+        }
+        if (cVar2 != '\0') {
+            EnableInterrupts();
+        }
+    }
+    else if (songIndex == STOP_FANFARE) {
+        cVar3 = DisableInterrupts();
+        if ((gFanfareMusicPlayerState != 0) && (gCurrentFanfareSong != STOP_SOUND_EFFECT)) {
+            if (IsMusicPlayerPlaying(INDEX_FANFARE)) {
+                m4aMPlayFadeOut(&gMPlayInfo_Fanfare,speed);
+            }
+            else {
+                m4aMPlayStop(&gMPlayInfo_Fanfare);
+                gCurrentFanfareSong = STOP_SOUND_EFFECT;
+            }
+        }
+        if (cVar3 != '\0') {
+            EnableInterrupts();
+        }
+    }
+    else if (IsFanfare(songIndex)) {
+        cVar4 = DisableInterrupts();
+        if ((gFanfareMusicPlayerState != 0) && (gCurrentFanfareSong == songIndex)) {
+            if (IsMusicPlayerPlaying(INDEX_FANFARE)) {
+                m4aMPlayFadeOut(&gMPlayInfo_Fanfare,speed);
+            }
+            else {
+                m4aMPlayStop(&gMPlayInfo_Fanfare);
+                gCurrentFanfareSong = STOP_SOUND_EFFECT;
+            }
+        }
+        if (cVar4 != '\0') {
+            EnableInterrupts();
+        }
+    }
+}
+
+bool8 IsFanfareSEPlaying(u16 songIndex)
+{
+  u32 playerIndex;
+  struct PMDMusicPlayer *musicPlayer;
+  register u32 songIndex_u32 asm("r4");
```

Comment:
Missing #NONMATCHING?

Reply:
This does match however, the main thing is the register pinning. Most of pokeemerald/pokefirered follow the same. My current philosophy is to use NONMATCHING when `asm_unified` is used.

Reply:
So I was wrong on the above (from GriffinR)
`fwiw, when pokeemerald/pokefirered still had uses of the register keyword they were wrapped in NONMATCHING`. Will update this function with `NONMATCHING` just to follow convention.